### ISSUE TITLE
Bad tokenization of C# nullable types

### ISSUE THREAD

**Issue by mihaipopescu** [uncrustify/uncrustify#1290]
Expected versus output diff:
---
```diff
diff --git "a/C:\\work\\external\\uncrustify\\tests\\output\\staging\\60033-UNI-21730.cs" "b/C:\\work\\external\\uncrustify\\tests\\results\\staging\\60033-UNI-21730.cs"
index 3dc1784..299de85 100644
--- "a/C:\\work\\external\\uncrustify\\tests\\output\\staging\\60033-UNI-21730.cs"
+++ "b/C:\\work\\external\\uncrustify\\tests\\results\\staging\\60033-UNI-21730.cs"
@@ -1,5 +1,5 @@
 private void Foo()
 {
-    Action<UnityPlayerBaseStartInfo, GraphicsTestRunConfiguration, Action<string, Bitmap, long>, RenderingBackend, DX11FeatureLevel? , string> playerRunnerImageCallback
+    Action<UnityPlayerBaseStartInfo, GraphicsTestRunConfiguration, Action<string, Bitmap, long>, RenderingBackend, DX11FeatureLevel ? , string> playerRunnerImageCallback
         = (playerStartInfo, description, incomingScreenshotCallback, configuration, dx11Featurelevel, graphicsDriverType) => incomingScreenshotCallback(filename, new Bitmap(1, 1), 42);
 }
```

Debug support files:
---
[60033-UNI-21730.cs](https://gist.githubusercontent.com/mihaipopescu/711c6fd5a36aa7158f4d516fcc3c7e15/raw/3dc17844133112a811b44491e4fdd4b6a3aaaa55/60033-UNI-21730.cs)
[60033-UNI-21730.cs.log](https://gist.githubusercontent.com/mihaipopescu/711c6fd5a36aa7158f4d516fcc3c7e15/raw/42f6ff2abe2e6ff386a29de54b6ceeec91eb4946/60033-UNI-21730.cs.log)
[60033-UNI-21730.cs.unc](https://gist.githubusercontent.com/mihaipopescu/711c6fd5a36aa7158f4d516fcc3c7e15/raw/a5002ae27619046d2ed2186b454de59998ed0577/60033-UNI-21730.cs.unc)
[Uncrustify.CSharp.cfg](https://gist.githubusercontent.com/mihaipopescu/711c6fd5a36aa7158f4d516fcc3c7e15/raw/1d064c1fabb76ce328dd015ed40023207de20ca1/Uncrustify.CSharp.cfg)
[UNI-21730.cs](https://gist.githubusercontent.com/mihaipopescu/711c6fd5a36aa7158f4d516fcc3c7e15/raw/3dc17844133112a811b44491e4fdd4b6a3aaaa55/UNI-21730.cs)

Disclaimer:
---
Note: This is a generated issue from our internal issue tracker.
Expect a dry description which could be insufficient to properly understand the issue.
On the other hand, the support files should be sufficient to easily reproduce and debug the issue,
but please comment if it's still unclear and I will get back with clarifications.
See main discussion thread here #1223.

**Comment by brendanlong** [comment]
I'm having the same problem in Vala, and there doesn't seem to be any option which convinces it to put the ? with the nullable type, without also removing space around the ternary operator.

**Comment by satishp87** [comment]
In unity fork this issue is fixed with Down-merge PR
https://github.com/Unity-Technologies/uncrustify/pull/48

**Comment by gmaurel** [comment]
Fixed at e0f021ae7d14e42f6d21aa3e9568f2ebd057ce41, Jan  6 18:07:41 2018
